File: anti-thief-new/app/src/main/java/com/emopass/antitheftalarm/adapter/LanguageAdapter.java
package com.emopass.antitheftalarm.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.appcompat.content.res.AppCompatResources;
import androidx.recyclerview.widget.RecyclerView;
import com.emopass.antitheftalarm.databinding.ItemLanguageViewBinding;
import com.emopass.antitheftalarm.model.LanguageModel;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class LanguageAdapter extends RecyclerView.Adapter<LanguageAdapter.ViewHolder> {
    private final Context context;
    private final List<LanguageModel> languageModelList = new ArrayList<>();
    private OnLanguageChange onLanguageChangedCallback;

    public LanguageAdapter(Context context) {
        this.context = context;
    }

    public void setData(List<LanguageModel> list) {
        languageModelList.clear();
        languageModelList.addAll(list);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemLanguageViewBinding binding = ItemLanguageViewBinding.inflate(
                LayoutInflater.from(parent.getContext()),
                parent,
                false
        );
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        LanguageModel language = languageModelList.get(position);
        holder.getBinding().imgLanguage.setImageDrawable(
                AppCompatResources.getDrawable(
                        context,
                        language.getResIcon()
                )
        );
        holder.getBinding().icChoose.setSelected(language.getChoose());
        holder.getBinding().ctlItemLanguage.setSelected(language.getChoose());
        holder.getBinding().txtLanguage.setText(context.getString(language.getLanguageName()));
        holder.itemView.setOnClickListener(view -> {
            updateView(language);
            onLanguageChangedCallback.onChange(itemSelected().getLanguageCode());
        });
    }

    @Override
    public int getItemCount() {
        return languageModelList.size();
    }

    private void updateView(LanguageModel languageModel) {
        for (LanguageModel model: languageModelList) {
            model.setChoose(Objects.equals(model.getLanguageCode(), languageModel.getLanguageCode()));
        }
        notifyDataSetChanged();
    }

    public LanguageModel itemSelected() {
        LanguageModel itemSelected = null;
        for (LanguageModel item: languageModelList) {
            if(item.getChoose()) {
                itemSelected = item;
                break;
            }
        }
        return  itemSelected;
    }

    public void setOnLanguageChange(OnLanguageChange callBack) {
        onLanguageChangedCallback = callBack;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private ItemLanguageViewBinding binding;
        public ViewHolder(ItemLanguageViewBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public ItemLanguageViewBinding getBinding() {
            return  binding;
        }
    }

    public interface OnLanguageChange {
        void onChange(String language);
    }
}

File: Wallpaper/app/src/main/java/com/parallax/hdvideo/wallpapers/ui/language/LanguageAdapter.kt
package com.parallax.hdvideo.wallpapers.ui.language

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.appcompat.content.res.AppCompatResources
import androidx.recyclerview.widget.RecyclerView
import com.parallax.hdvideo.wallpapers.data.model.LanguageModel
import com.parallax.hdvideo.wallpapers.databinding.ItemLanguageViewBinding

class LanguageAdapter(private val context: Context) :
    RecyclerView.Adapter<LanguageAdapter.ViewHolder>() {
    private var languageModelList: MutableList<LanguageModel> = mutableListOf()
    private var onLanguageChangedCallback: (String) -> Unit = {}
    fun setData(data: List<LanguageModel>) {
        languageModelList.clear()
        languageModelList.addAll(data)
        notifyDataSetChanged()
    }

    class ViewHolder(itemViewLanguageBinding: ItemLanguageViewBinding) :
        RecyclerView.ViewHolder(itemViewLanguageBinding.root) {
        val itemBinding = itemViewLanguageBinding
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding =
            ItemLanguageViewBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    override fun getItemCount(): Int {
        return languageModelList.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val language = languageModelList[position]
        holder.itemBinding.imgLanguage.setImageDrawable(
            AppCompatResources.getDrawable(
                context,
                language.idResIcon
            )
        )
        holder.itemBinding.icChoose.isSelected = language.isChosen
        holder.itemBinding.ctlItemLanguage.isSelected = language.isChosen
        holder.itemBinding.txtLanguage.text = context.getString(language.languageName)
        holder.itemView.setOnClickListener {
            updateView(language)
            onLanguageChangedCallback(itemSelected().languageCode)
        }
    }

    private fun updateView(languageModel: LanguageModel) {
        languageModelList.forEach {
            it.isChosen = it.languageCode == languageModel.languageCode
        }
        notifyDataSetChanged()
    }

    fun itemSelected(): LanguageModel {
        return languageModelList.first { it.isChosen }
    }

    fun setOnLanguageChange(handle: (String) -> Unit) {
        onLanguageChangedCallback = handle
    }
}